kb/2025-10-07/troubleshooting_vm_scheduling_issues_nodeselector.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Troubleshooting VM Live Migration Issues Caused by Node Selectors
+description: 'How to resolve VM live migration failures when target nodes lack specific features due to nodeSelector constraints.'
+slug: troubleshooting_vm_scheduling_issues_nodeselector
+authors:
+  - name: Jack Yu
+    title: Senior Software Engineer
+    url: https://github.com/Yu-Jack
+    image_url: https://github.com/Yu-Jack.png
+tags: [harvester, virtual machine, VM, scheduling, nodeSelector, troubleshooting]
+hide_table_of_contents: false
+---
+
+## Problem Description
+
+When a virtual machine (VM) cannot be live migrated to a target node because of `nodeSelector` constraints, it often indicates a mismatch between the VM's requirements and the node's capabilities.
+
+For example, a VM might have a `nodeSelector` requiring a specific CPU feature (e.g., `cpu-feature.node.kubevirt.io/fpu: "true"`), but the target node lacks this label. This prevents the scheduler from migrating the VM to that node.
+
+This article describes how to resolve this issue using two different approaches.
+
+## Solution 1: Reboot the VM
+
+If the `nodeSelector` was automatically added by KubeVirt (for instance, during a previous migration or initial start), rebooting the VM will clear these dynamic selectors.
+
+Once the VM is rebooted, the restrictive `nodeSelector` is removed, allowing the VM to be scheduled on any available node.
+
+To ensure the VM can migrate to other nodes more easily in the future without encountering similar issues, refer to [Setting up a common CPU model for virtual machine migration](../2025-09-30/setup_common_cpu_model_for_vm_live_migration.md).
```

Comment:
> should we also include the case of choosing a specific cpuModel that can be satisfied across the cluster and using it a per VM or cluster level via the kubevirt config update

@ibrokethecloud Do you think that it's enough? Or should we have a isolated section about this in the markdown?